source/cppfront.cpp

```diff
@@ -226,7 +226,13 @@ class positional_printer
 
         //  Not using print() here because this is transparent to the curr_pos
         if (!flag_clean_cpp1) {
-            out << "#line " << line << " \"" << filename << "2\"\n";
+            out << "#line " << line << " \"";
+            for (auto c : filename)
+            {
+               if (c == '\\') out << "\\\\";
+               else out << c;
+            }
+            out << "2\"\n";
```

Comment:
Could this work instead?
```suggestion
            const std::string filename2 = filename + '2';
            out << "#line " << line << ' ' << std::quoted(filename2) << '\n';
```
Or maybe `filename` should be renamed to `filename2` and have `'2'` appended, and `filename()` be a new member function that returns a `std::string_view` into `filename2` without the `'2'`.